pkg/detector/detector.go

```diff
@@ -1026,19 +1026,34 @@ func (d *ResourceDetector) ReconcileResourceBinding(key util.QueueKey) error {
 	}
 
 	klog.Infof("Reconciling resource binding(%s/%s)", binding.Namespace, binding.Name)
-	switch binding.Spec.Resource.Kind {
-	case util.DeploymentKind:
-		return d.AggregateDeploymentStatus(binding.Spec.Resource, binding.Status.AggregatedStatus)
-	case util.ServiceKind:
-		return d.AggregateServiceStatus(binding.Spec.Resource, binding.Status.AggregatedStatus)
-	case util.IngressKind:
-		return d.AggregateIngressStatus(binding.Spec.Resource, binding.Status.AggregatedStatus)
-	case util.JobKind:
-		return d.AggregateJobStatus(binding.Spec.Resource, binding.Status.AggregatedStatus, binding.Spec.Clusters)
-	default:
-		// Unsupported resource type.
+	resource := binding.Spec.Resource
+	gvr, err := restmapper.GetGroupVersionResource(
+		d.RESTMapper, schema.FromAPIVersionAndKind(resource.APIVersion, resource.Kind),
+	)
+	if err != nil {
+		klog.Errorf("Failed to get GVR from GVK %s %s. Error: %v", resource.APIVersion, resource.Kind, err)
+		return err
+	}
+	obj, err := d.DynamicClient.Resource(gvr).Namespace(resource.Namespace).Get(context.TODO(), resource.Name, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get resource(%s/%s/%s), Error: %v", resource.Kind, resource.Namespace, resource.Name, err)
+		return err
+	}
+
+	newObj, err := d.ResourceInterpreter.AggregateStatus(obj, binding.Status.AggregatedStatus, binding.Spec.Clusters)
+	if err != nil {
+		klog.Errorf("AggregateStatus for resource(%s/%s/%s) failed: %v", resource.Kind, resource.Namespace, resource.Name, err)
+		return err
+	}
+	if newObj == nil {
 		return nil
 	}
+
+	if _, err = d.DynamicClient.Resource(gvr).Namespace(resource.Namespace).UpdateStatus(context.TODO(), newObj, metav1.UpdateOptions{}); err != nil {
```

Comment:
The update may be failed, we can add `retry.RetryOnConflict`.

Reply:
It's a little tricky to use `retry` here, we need to retrieve the object, execute `AggregateStatus` and update it again, which would be a lot of redundant code, perhaps it would be better to return the error directly to trigger its resync.

Reply:
Perhaps we can directly copy the status part with the get newest object?